ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/CommandSubmitterITSpec.scala

```diff
@@ -0,0 +1,259 @@
+    }
+
+    val avgSizeOfNonconsumingExercise: Int = {
+      if (nonconsumingExercises.isEmpty) 0
+      else
+        nonconsumingExercises.map(_.choiceArgumentsSerializedSize).sum / nonconsumingExercises.size
+    }
+
+    val numberOfCreatesPerTemplateName: Map[String, Int] = {
+      val groups = createEvents.groupBy(_.templateName)
+      expectedTemplateNames.map(name => name -> groups.get(name).fold(0)(_.size)).toMap
+    }
+
+    val avgSizeOfCreateEventPerTemplateName: Map[String, Int] = {
+      val groups = createEvents.groupBy(_.templateName)
+      expectedTemplateNames.map { name =>
+        val avgSize = groups
+          .get(name)
+          .fold(0)(events =>
+            if (events.isEmpty) 0 else events.map(_.createArgumentsSerializedSize).sum / events.size
+          )
+        name -> avgSize
+      }.toMap
+    }
+
+  }
+
+}
+
+/** Collects information about create and exercise events.
+  */
+class EventsObserver(expectedTemplateNames: Set[String], logger: Logger)
+    extends ObserverWithResult[GetTransactionTreesResponse, ObservedEvents](logger) {
+
+  private val createEvents = collection.mutable.ArrayBuffer[ObservedCreateEvent]()
+  private val exerciseEvents = collection.mutable.ArrayBuffer[ObservedExerciseEvent]()
+
+  override def streamName: String = "dummy-stream-name"
+
+  override def onNext(value: GetTransactionTreesResponse): Unit = {
+    value.transactions.foreach { transaction: TransactionTree =>
+      val rootEvents = transaction.rootEventIds.map(transaction.eventsById)
+      rootEvents.foreach { event: TreeEvent =>
+        event.kind.created.foreach { created: CreatedEvent =>
+          val argsSize = created.createArguments.fold(0)(_.serializedSize)
+          val templateName =
+            created.templateId.getOrElse(sys.error(s"Expected templateId in $created")).entityName
+          createEvents.addOne(
+            ObservedCreateEvent(
+              templateName = templateName,
+              createArgumentsSerializedSize = argsSize,
+            )
+          )
+        }
+        event.kind.exercised.foreach { exercised =>
+          val argsSize = exercised.choiceArgument.fold(0)(_.serializedSize)
+          val templateName = exercised.templateId
+            .getOrElse(sys.error(s"Expected templateId in $exercised"))
+            .entityName
+          val choiceName = exercised.choice
+          exerciseEvents.addOne(
+            ObservedExerciseEvent(
+              templateName = templateName,
+              choiceName = choiceName,
+              choiceArgumentsSerializedSize = argsSize,
+              consuming = exercised.consuming,
+            )
+          )
+        }
+      }
+    }
+  }
+
+  override def completeWith(): Future[ObservedEvents] = Future.successful(
+    ObservedEvents(
+      expectedTemplateNames = expectedTemplateNames,
+      createEvents = createEvents.toList,
+      exerciseEvents = exerciseEvents.toList,
+    )
+  )
+}
+
+class CommandSubmitterITSpec
```

Comment:
Can you please move this class first in the file?

Reply:
Ok